pkg/controller/dnszone/dnszone_controller.go

```diff
@@ -204,6 +219,19 @@ func (r *ReconcileDNSZone) Reconcile(ctx context.Context, request reconcile.Requ
 		}
 
 		dnsLog.WithError(err).Error("error instantiating actuator")
+		var changed bool
+		desiredState.Status.Conditions, changed = controllerutils.SetDNSZoneConditionWithChangeCheck(
+			desiredState.Status.Conditions,
+			hivev1.GenericDNSErrorsCondition,
+			corev1.ConditionTrue,
+			"ActuatorNotInitialized",
+			"error instantiating actuator: "+controllerutils.ErrorScrub(err),
+			controllerutils.UpdateConditionIfReasonOrMessageChange)
+		if changed {
+			if err := r.Status().Update(context.Background(), desiredState); err != nil {
+				dnsLog.WithError(err).Log(controllerutils.LogLevel(err), "failed to update dnszone status")
+			}
+		}
```

Comment:
Same here, we need to be sure to clear this when everything is ok. If i've missed it then my bad. :)

Reply:
Same as above - so reconcileDNSProvider is called, and then setErrorConditions checks if there was an error, otherwise sets those error conditions to false. That code block will always execute for a successful reconcile, so I kept the code there in order to limit the update events